elephant/conversion.py

```diff
@@ -152,3 +155,704 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+                # Append the column index for each spike
+                l.extend([col] * count)
+            spike_idx.append(l)
+        return spike_idx
+
+    @property
+    def mat_clipped(self):
+        """
+        Returns a sparse matrix (`scipy.sparse.csr_matrix`), which rows
+        represent the number of spike trains and the columns represent the
+        binned index position of a spike in a spike train.
+        The matrix columns contain only ones, which indicate a spike.
+
+        Returns
+        -------
+        clipped matrix : numpy.ndarray
+            Returns a dense matrix representation of the sparse matrix,
+            with ones indicating a spike and zeros for non spike.
+            The ones in the columns represent the index
+            position of the spike in the spike train and rows represent the
+            number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> import quantities as pq
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.Binned(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.mat_clipped)
+        [[1 1 0 1 1 1 1 0 0 0]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+        """
+        return abs(scipy.sign(self.mat_unclipped))
+
+    @property
+    def mat_unclipped(self):
+        """
+        Returns the sparse matrix, which rows represents the number of
+        spike trains and the columns represents the binned index position
+        of a spike in a spike train.
+        The  matrix columns contain the number of spikes that
+        occurred in the spike train(s).
+        If **bool** :attr:`store_mat` is set to **True** last calculated
+        `unclipped` matrix will be returned.
+
+        Returns
+        -------
+        unclipped matrix : numpy.ndarray
+            Matrix with spike times. Columns represent the index position of
+            the binned spike and rows represent the number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.Binned(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.mat_unclipped)
+        [[2 1 0 1 1 1 1 0 0 0]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+
+        """
+        if self.mat_u is not None:
+            return self.mat_u
+        if self.store_mat_u:
+            self.mat_u = self.sparse_mat_unclip.toarray()
+            return self.mat_u
+        # Matrix on demand
+        else:
+            return self._sparse_mat_u.toarray()
+
+    def store_mat_unclipped(self):
```

Comment:
This is public API but undocumented.

More fundamentally, you have a public attribute `store_mat_u`, which if changed does absolutely nothing.  I would think it would be better to have this be a property, where the setter is used to store or erase the matrix.  Then `mat_u` should be made private.  Then `mat_unclipped` can just check if `mat_u` is None.  That way, changing the value has immediate effects.  

The same could be done for `store_sparse_mat_clip`.